xdsl/dialects/wasm/ops.py

```diff
@@ -73,10 +168,30 @@ def encode(self, ctx: WasmBinaryEncodingContext, io: BinaryIO) -> None:
         io.write(magic)
         io.write(version)
 
+        # FIXME: implement encoder for module attributes
+        if (
+            len(self.tables.data)
+            + len(self.mems.data)
+            + len(self.exports)
+            + len(self.imports)
+            != 0
+        ):
+            raise NotImplementedError()
```

Comment:
Could you please add a diagnostic filecheck test?
```suggestion
            raise DiagnosticEception("Module attributes printing not implemented yet")
```
SImilarly for the notimplemented below

Reply:
What is a diagnostic filecheck test?

Reply:
We have a few around, it's a way to test that a certain error is raised
```
// RUN: xdsl-opt %s --verify-diagnostics | filecheck %s
```